pulp_docker/app/models.py

```diff
@@ -1,37 +1,177 @@
-"""
-Check `Plugin Writer's Guide`_ for more details.
-
-.. _Plugin Writer's Guide:
-    http://docs.pulpproject.org/en/3.0/nightly/plugins/plugin-writer/index.html
-"""
-
 from logging import getLogger
+from types import SimpleNamespace
+
+from django.db import models
 
-from django.db import models # noqa
+from pulpcore.plugin.models import Content, Remote, Publisher
 
-from pulpcore.plugin.models import Content, ContentArtifact, Remote, Publisher  # noqa
 
 logger = getLogger(__name__)
 
 
-class DockerContent(Content):
+MEDIA_TYPE = SimpleNamespace(
+    MANIFEST_V1='application/vnd.docker.distribution.manifest.v1+json',
+    MANIFEST_V2='application/vnd.docker.distribution.manifest.v2+json',
+    MANIFEST_LIST='application/vnd.docker.distribution.manifest.list.v2+json',
+    CONFIG_BLOB='application/vnd.docker.container.image.v1+json',
+    REGULAR_BLOB='application/vnd.docker.image.rootfs.diff.tar.gzip',
+    FOREIGN_BLOB='application/vnd.docker.image.rootfs.foreign.diff.tar.gzip',
+)
+
+
+class Manifest(Content):
+    """
+    A docker manifest.
+
+    This content has no artifacts.
+
+    Fields:
+        digest (models.CharField): The manifest digest.
+        schema_version (models.IntegerField): The docker schema version.
+        media_type (models.CharField): The manifest media type.
     """
-    The "docker" content type.
 
-    Define fields you need for your new content type and
-    specify uniqueness constraint to identify unit of this type.
+    TYPE = 'manifest'
+
+    digest = models.CharField(max_length=255)
+    schema_version = models.IntegerField()
+    media_type = models.CharField(
+        max_length=60,
+        choices=(
+            (MEDIA_TYPE.MANIFEST_V1, MEDIA_TYPE.MANIFEST_V1),
+            (MEDIA_TYPE.MANIFEST_V2, MEDIA_TYPE.MANIFEST_V2),
+        ))
 
-    For example::
+    class Meta:
+        unique_together = ('digest',)
 
-        field1 = models.TextField()
-        field2 = models.IntegerField()
-        field3 = models.CharField()
 
-        class Meta:
-            unique_together = (field1, field2)
+class ManifestBlob(Content):
     """
+    A blob defined within a manifest.
 
-    TYPE = 'docker'
+    The actual blob file is stored as an artifact.
+
+    Fields:
+        digest (models.CharField): The blob digest.
+        media_type (models.CharField): The blob media type.
+
+    Relations:
+        manifest (models.ForeignKey): Many-to-one relationship with Manifest.
+    """
+
+    TYPE = 'manifest-blob'
+
+    digest = models.CharField(max_length=255)
+    media_type = models.CharField(
```

Comment:
@asmacdo  when you will be implementing sync, at the parsing of the manifest.json: if the information for the mediatype of the blob will be available, then use that info, otherwise set the value to MEDIA_TYPE.REGULAR_BLOB
** for the background, in schema1 there is no information available for blobs, just its digest. BUt also schedma 1 can copy only with application/vnd.docker.image.rootfs.diff.tar.gzip mediatype